### PR TITLE
Desc table_name interface adapt to hive external table (#1197)

### PR DESCRIPTION
**Pull request by Ielihs**
before:
```
mysql> desc external_hive_tbl all;
ERROR 1064 (HY000): Unknown storage engine 'HIVE'
```

after:
```
mysql> desc external_hive_tbl;
+-------+------------+------+------+---------+-------+
| Field | Type       | Null | Key  | Default | Extra |
+-------+------------+------+------+---------+-------+
| id    | BIGINT     | Yes  | true | NULL    |       |
| name  | VARCHAR(1) | Yes  | true | NULL    |       |
| city  | VARCHAR(1) | Yes  | true | NULL    |       |
+-------+------------+------+------+---------+-------+
3 rows in set (0.01 sec)
```

cc @gengjun-git

### CODE REVIEW

**Comment by imay** [comment]
@Ielihs 
Thanks for your contribution.

File: fe/fe-core/src/main/java/com/starrocks/analysis/DescribeStmt.java [hunk 1]
```diff
@@ -247,6 +248,28 @@ public void analyze(Analyzer analyzer) throws AnalysisException, UserException {
                             mysqlTable.getMysqlDatabaseName(),
                             mysqlTable.getMysqlTableName());
                     totalRows.add(row);
+                } else if (table.getType() == TableType.HIVE) {
+                    isOlapTable = true;
```
Comment (imay) [inline fe/fe-core/src/main/java/com/starrocks/analysis/DescribeStmt.java:252]: Hive Table is not OLAP Table, this should be false.

**Review by imay — DISMISSED** [review]

**Comment by Ielihs** [comment]
@imay PTAL

**Review by imay — APPROVED** [review]

File: fe/fe-core/src/main/java/com/starrocks/analysis/DescribeStmt.java [hunk 2]
```diff
@@ -166,6 +166,11 @@ public void analyze(Analyzer analyzer) throws AnalysisException, UserException {
                 ErrorReport.reportAnalysisException(ErrorCode.ERR_BAD_TABLE_ERROR, dbTableName.getTbl());
             }
 
+            if (table.getType() == TableType.HIVE) {
+                // Reuse the logic of `desc <table_name>` because hive external table doesn't support view.
+                isAllTables = false;
```
Comment (imay) [inline fe/fe-core/src/main/java/com/starrocks/analysis/DescribeStmt.java:171]: I think it is better to use a local variable other than changing the member variable.

Comment (Ielihs) [reply fe/fe-core/src/main/java/com/starrocks/analysis/DescribeStmt.java:171]: The member variables here are only used as markers, and the effect should be controllable.

**Review by gengjun-git — APPROVED** [review]

**Pull request merged**